test/views/declaration/CommodityDetailsViewSpec.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.declaration
+
+import forms.declaration.CommodityDetails
+import helpers.views.declaration.CommonMessages
+import models.DeclarationType.DeclarationType
+import models.{DeclarationType, Mode}
+import org.jsoup.nodes.Document
+import services.cache.ExportsTestData
+import unit.tools.Stubs
+import views.declaration.spec.UnitViewSpec
+import views.html.declaration.commodity_details
+import views.tags.ViewTest
+
+@ViewTest
+class CommodityDetailsViewSpec extends UnitViewSpec with ExportsTestData with Stubs with CommonMessages {
+
+  private val page = new commodity_details(mainTemplate, minimalAppConfig)
+  private val itemId = "item1"
+  private val realMessages = validatedMessages
+  private def createView(mode: Mode = Mode.Normal, itemId: String = itemId, declarationType: DeclarationType = DeclarationType.STANDARD): Document =
+    page(mode, itemId, CommodityDetails.form(declarationType))(journeyRequest(), realMessages)
+
+  def commodityDetailsView(declarationType: DeclarationType): Unit = {
```

Comment:
Why this method contains whole specs inside? I can't find where is the closing bracket. But it looks like everything is in this method and this method is not invoked. View is created inside.

Reply:
This method is called by the three specs at the bottom of the file - once for each dec type.   Its a way of parameterising the tests that Lukasz showed me